source/kraken/realtime.cpp

```diff
@@ -198,25 +226,12 @@ create_disruption(const std::string& id,
                         departure_time = arrival_time =st.arrival().time();
                     }
                 }
-                auto first_day_of_impact = get_first_day_of_imapct(*impact);
-                auto arrival_seconds = boost::posix_time::from_time_t(arrival_time).time_of_day().total_seconds();
-                auto departure_seconds = boost::posix_time::from_time_t(departure_time).time_of_day().total_seconds();
-                
-                // If the arrival/depature time's date is later than the first day of impact, this means
-                // the VJ passes the midnight, it's pickup/dropoff time should add 86400 seconds(one day)
-                // Ex:
-                // the first day of impact is 1st, Jan, the stoptime's base arrival_time is 23:50 on 1st Jan, 
-                // and the new arrival_time is tommorow monrning at 00:10. The new arrival is computed as
-                // 10*60 + 86400 seconds.
-                auto get_diff_days = [&](uint32_t time){
-                    return (boost::posix_time::from_time_t(time).date() - first_day_of_impact).days();
-                };
-
-                arrival_seconds +=  get_diff_days(arrival_time) * navitia::DateTimeUtils::SECONDS_PER_DAY;
-                departure_seconds += get_diff_days(departure_time) * navitia::DateTimeUtils::SECONDS_PER_DAY;
+                auto ptime_arrival = bpt::from_time_t(arrival_time) - start_first_day_of_impact;
+                auto ptime_departure = bpt::from_time_t(departure_time) - start_first_day_of_impact;
 
-                type::StopTime stop_time{static_cast<uint32_t>(arrival_seconds),
-                    static_cast<uint32_t>(departure_seconds), stop_point_ptr};
+                type::StopTime stop_time{static_cast<uint32_t>(ptime_arrival.total_seconds()),
+                                         static_cast<uint32_t>(ptime_departure.total_seconds()),
```

Comment:
I prefere using the constructor syntax for that (`uint32_t(ptime_arrival.total_seconds())`) but do as you want.